Vagrantfile

```diff
@@ -43,6 +66,7 @@ Vagrant.configure("2") do |config|
     SHELL
 
     centos.vm.synced_folder ".", "/vagrant", type: "rsync"
+    centos.vm.box_check_update = false
```

Comment:
I tried to summarize the behaviors in my previous comment https://github.com/retis-org/retis/pull/231#discussion_r1348510124

ATM, it seems we don't update the boxes that are not rawhide, and this is the takeaway from this PR.
It is out of the scope of this PR, so it was not included/fixed here.

Reply:
I think one option would be to add to run up (before `vagrant up`) `vagrant box update && vagrant box prune` adding in cirrus `reupload_on_changes`.
Not 100% sure this would not create any side effect (useless files and so forth) but might worth a try.

Reply:
> That might work and be better from logical point of view (all logic done in Cirrus). Also with that we could use `upload_caches` to upload the box images before the end of the task to maximize cache availability (when CI runs in parallel).
> 
> My only concern would be how `fingerprint_script` works with `reupload_on_changes`.

initially `reupload_on_changes` defaulted to true when fingerprint_script was present, and then they decided to turn it to false.
With both set, the cache is regenerated if the hash of the output of fingerprint_script (new hash key) and updated (same hash key as the old fingerprint_script) if the content changes.

What I don't know is if `upload_caches` just uploads in case there's a difference (as one would expect) so honoring `reupload_on_changes`.

Reply:
re-run f38 and rawhide. F38 was still cached, while rawhide got updated (as expected). One more re-run of rawhide resulted correctly in a cache hit.
Based on the above, the cache now works as intended.